### PR TITLE
Document all options for protect_from_forgery.

### LINKED ISSUE #18329: Enhance protect_from_forgery with inline skipping
Currently we want you to do this to only partially apply the forgery protection:

```ruby
class FooController < ApplicationController
  protect_from_forgery except: :index
  skip_before_action :verify_authenticity_token, if: -> { request.format.json? }
end
```
That's needlessly meddling with the forgery implementation. Let's just do this:

```ruby
class FooController < ApplicationController
  protect_from_forgery except: :index, unless: -> { request.format.json? }
end
```

The unless argument can be either a symbol reference to a method or a proc.

### PR DESCRIPTION
**Pull request by simi**
Related to #18329.

/cc @dhh

### CODE REVIEW

**Comment by dhh** [comment]
Need to use this in the example instead of before_skip_action :verify_authentication_token.

**Comment by simi** [comment]
Yup I've noticed right now. Force pushed.

File: actionpack/lib/action_controller/metal/request_forgery_protection.rb [hunk 1]
```diff
@@ -29,8 +29,7 @@ class InvalidCrossOriginRequest < ActionControllerError #:nodoc:
   # you're building an API you'll need something like:
   #
   #   class ApplicationController < ActionController::Base
-  #     protect_from_forgery
-  #     skip_before_action :verify_authenticity_token, if: :json_request?
+  #     protect_from_forgery unless: json_request?
```
Comment (dhh) [inline actionpack/lib/action_controller/metal/request_forgery_protection.rb]: Needs to be `unless: :json_request?` (missing the symbol).

Comment (dhh) [inline actionpack/lib/action_controller/metal/request_forgery_protection.rb]: And while we're at it, let's inline it: `protect_from_forgery unless: -> { request.format.json? }`

**Comment by simi** [comment]
Done @dhh. I also changed documentation format. Not is is one-liner since same format is used for `only/except` option.

**Pull request merged**